src/darwin/Framework/CHIPTests/MTRBackwardsCompatTests.m

```diff
@@ -94,24 +91,44 @@ - (void)onCommissioningComplete:(NSError *)error
 @interface MTRBackwardsCompatTests : XCTestCase
 @end
 
+static BOOL sStackInitRan = NO;
+static BOOL sNeedsStackShutdown = YES;
+
 @implementation MTRBackwardsCompatTests
 
++ (void)tearDown
+{
+    // Global teardown, runs once
+    if (sNeedsStackShutdown) {
+        // We don't need to worry about ResetCommissionee.  If we get here,
+        // we're running only one of our test methods (using
+        // -only-testing:MatterTests/MTROTAProviderTests/testMethodName), since
+        // we did not run test999_TearDown.
+        [self shutdownStack];
+    }
+}
+
 - (void)setUp
 {
+    // Per-test setup, runs before each test.
     [super setUp];
     [self setContinueAfterFailure:NO];
+
+    if (sStackInitRan == NO) {
```

Comment:
Do we need a global lock plus a dispatch_once to wait for the first `initStack` finishing or will individual tests automagically "just work" when their operations properly process after one `initStack` finishes?

Reply:
initStack is blocking; it does not return until it finishes.

And after it has run once, it does not need to get run again.

Reply:
Okay I was asking this, thinking about parallel unit test running but seems like the scheme has "Execute in parallel" off for now, and so that makes this point moot. Thanks you.